grin/grin.py

```diff
@@ -412,7 +412,15 @@ def main(argv=None):
         g = GrepText(regex, args)
         openers = dict(text=open, gzip=gzip.open)
         for filename, kind in get_filenames(args):
-            report = g.grep_a_file(filename, opener=openers[kind], encoding=args.encoding)
+            try:
+                report = g.grep_a_file(filename, opener=openers[kind], encoding=args.encoding)
+            except (gzip.BadGzipFile, EOFError) as ex:
+                if kind != "gzip":
+                    # probably shouldn't happen; something weird
+                    raise
+                # Alert the user, or just silently switch to plaintext grep?
+                print(f"{filename}: {ex}; retrying as plaintext.", file=sys.stderr)
```

Comment:
`f-string` are python >= 3.6, and 2.6.X supports 3.5.
I've anyway removed the alert, so no real problem.